flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java

```diff
@@ -192,7 +194,8 @@ public RocksDBKeyedStateBackendBuilder(
 			ttlTimeProvider,
 			metricGroup,
 			stateHandles,
-			keyGroupCompressionDecorator
+			keyGroupCompressionDecorator,
+			cancelStreamRegistry
```

Comment:
Pleas notice that this is also a semantic change of the this registry: I think before when we passed in `cancelStreamRegistry`, the scope was only for controlling streams and closeables during restore. The backend itself was creating it's own `CloseableRegistry` object, which is not the same as the `cancelStreamRegistry` used during the restore. The difference is that they have different scopes and could be closed individually, e.g. you could close the `cancelStreamRegistry` after the restore, but now this would also close the backend's main registry. This leads to a small leak, because `cancelStreamRegistry` is never unregistered. So closing the backend is invoked twice as well, once through this registry that is never unregistered from the task and the other one because we register the backend itself as a closeable to the task, see `BackendRestorerProcedure::attemptCreateAndRestore`, L148. And the matching unregister is `AbstractStreamOperator::dispose`, e.g. L345. So that registry is effectively closed from two ends.

Reply:
> I think before when we passed in cancelStreamRegistry, the scope was only for controlling streams and closeables during restore.

Got the point. Some minor correction: we don't pass in any registry before, and the reason we don't have the issue described in FLINK-11804 is that the invocation of `backendInstance.restore` was surrounded by a try-catch clause and the whole `backendInstance` will be unregistered from the task registry and then disposed, thus the `CloseableRegistry` created inside the backend instance will also be closed during the backend's `dispose`. However, after we move restore operation into backend's building process, we need to separate the `cancelStreamRegistry` used in restore and backend constructing, and make sure the one used in restore is registered and handled by the task cancel registry.